### PR TITLE
Add ti attributes to react ti details

### PR DESCRIPTION
**Pull request by bbovenzi**
Part of moving everything to React. Show all Task Instance attributes info in the Task Instance details panel of the grid view:

Screenshot 2024-02-08 at 4 24 54 PM

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -3402,6 +3405,123 @@ def graph_data(self, session: Session = NEW_SESSION):
...
+    @expose("/object/task_instance_attributes")
+    @auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)
+    @gzipped
+    @action_logging
+    @provide_session
+    def task_instance_attributes(self, session: Session = NEW_SESSION):
+        """Retrieve task instance attributes."""
+        dag_id = request.args.get("dag_id")
+        task_id = request.args.get("task_id")
+        run_id = request.args.get("run_id")
```
Comment (jedcunningham) [inline airflow/www/views.py:3417]: missing map_index

File: airflow/www/views.py [hunk 2]
```diff
@@ -3402,6 +3405,123 @@ def graph_data(self, session: Session = NEW_SESSION):
...
+            ti_attrs: list[tuple[str, Any]] | None = None
+        else:
+            ti.refresh_from_task(task)
+            ti_attrs_to_skip = [
+                "dag_id",
+                "dag_model",
+                "dag_run",
+                "run_id",
+                "task_id",
+                "task_instance_notedag_run",
```
Comment (jedcunningham) [inline airflow/www/views.py:3448]:
```suggestion
                "dag_run",
```
I assume we want the ti note.

File: airflow/www/views.py [hunk 3]
```diff
@@ -3402,6 +3405,123 @@ def graph_data(self, session: Session = NEW_SESSION):
...
+                "task",
+                "trigger",
+                "triggerer_job",
+                "try_number",
+                "previous_ti",
+                "previous_ti_success",
+                "previous_start_date_success",
+                "queued_dttm",
+            ]
+            # Some fields on TI are deprecated, but we don't want those warnings here.
```
Comment (jedcunningham) [inline airflow/www/views.py:3470]: We probably shouldn't include deprecated fields?

File: airflow/www/views.py [hunk 4]
```diff
@@ -3402,6 +3405,123 @@ def graph_data(self, session: Session = NEW_SESSION):
...
+            ti.refresh_from_task(task)
+            ti_attrs_to_skip = [
+                "dag_id",
+                "dag_model",
+                "dag_run",
+                "run_id",
+                "task_id",
+                "task_instance_notedag_run",
+                "duration",
+                "start_date",
```
Comment (jedcunningham) [inline airflow/www/views.py:3450]: Why don't we want these? Since we already have them?

File: airflow/www/views.py [hunk 5]
```diff
@@ -3402,6 +3405,123 @@ def graph_data(self, session: Session = NEW_SESSION):
...
+            # Some fields on TI are deprecated, but we don't want those warnings here.
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", RemovedInAirflow3Warning)
+                all_ti_attrs = (
+                    # fetching the value of _try_number to be shown under name try_number in UI
+                    (name, getattr(ti, "_try_number" if name == "try_number" else name))
+                    for name in dir(ti)
+                    if not name.startswith("_") and name not in ti_attrs_to_skip
+                )
+            ti_attrs = sorted((name, attr) for name, attr in all_ti_attrs if not callable(attr))
```
Comment (jedcunningham) [inline airflow/www/views.py:3479]: You could check callable in the loop avoid to avoid looping again.

File: airflow/www/views.py (hunk 4 above)
Comment (bbovenzi) [reply airflow/www/views.py:3450]: Exactly. This is basically just filling in for what's missing in the API. But I don't know what we want all of it in the rest api.

**Comment by bbovenzi** [comment]
Gonna try to do this all via the rest api instead

**Pull request closed without merging**